### DISCUSSION TITLE
CheckOAuthRedirectionActivation() not available on WinUIEx.WebAuthenticator. Using 1.5 version of the nuget package.

### DISCUSSION THREAD

**Discussion by ataparia** [dotMorten/WinUIEx#138] [General]
Hi Team,
I don't see any CheckOAuthRedirectionActivation() method on WinUIEx.WebAuthenticator class. I am usin g1.5 version. What am I missing? 
I only see AuthenticateAsync method on that class.

**Comment by dotMorten** [comment]
That method was introduced in v2.2. Earlier versions didn't need this, but the approach used before caused a lot of problems and crashes.
Why still 1.5?

**Reply by ataparia** [reply]
My project doesn't compile when I use the latest versions.
1. With v2.2 I get the following errors:
-------------------------------------------
error NU1605: Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.3.230502000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.2.0 -> Microsoft.WindowsAppSDK (>= 1.3.230502000) 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)
Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.3.230502000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.2.0 -> Microsoft.WindowsAppSDK (>= 1.3.230502000) 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)
2>Done building project "Infor.PublicSector.Mobile.FieldWorks.csproj" -- FAILED.
error NU1605: Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.3.230502000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.2.0 -> Microsoft.WindowsAppSDK (>= 1.3.230502000) 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)
error NU1605: Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.3.230502000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.2.0 -> Microsoft.WindowsAppSDK (>= 1.3.230502000) 
 error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)

With v2.3, I get the following errors:
--------------------------------------------------------------------
error NU1605: Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.4.230822000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.3.0 -> Microsoft.WindowsAppSDK (>= 1.4.230822000) 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)
2>Done building project "Infor.PublicSector.Mobile.FieldWorks.csproj" -- FAILED.
error NU1605: Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.4.230822000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.3.0 -> Microsoft.WindowsAppSDK (>= 1.4.230822000) 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)
2>Done building project "Infor.PublicSector.Mobile.FieldWorks.csproj" -- FAILED.
error NU1605: Warning As Error: Detected package downgrade: Microsoft.WindowsAppSDK from 1.4.230822000 to 1.2.221209.1. Reference the package directly from the project to select a different version. 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> WinUIEx 2.3.0 -> Microsoft.WindowsAppSDK (>= 1.4.230822000) 
error NU1605:  Infor.PublicSector.Mobile.FieldWorks -> Microsoft.WindowsAppSDK (>= 1.2.221209.1)

**Reply by dotMorten** [reply]
Please see the section on how to use it with .NET MAUI: https://dotmorten.github.io/WinUIEx/concepts/Maui.html#include-winuiex-package-in-windows-target

**Reply by ataparia** [reply]
Yes, I went over that article and that's when I was using 1.5 version.

This is how I am using it in my project file:
PackageReference Include="WinUIEx" Version="2.3" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'"

**Reply by dotMorten** [reply]
And did you do the bit about explicitly including `Microsoft.WindowsAppSDK` v1.4 ?

**Reply by ataparia** [reply]
I didn't do that part. I don't know why but I thought that the article talks about correcting the version of the WinUIEx library itself instead of adding the Microsoft.WindowsAppSDK. I am really sorry about this. Thanks again for your help and pointing me in the right direction.
